pkg/server/server.go

```diff
@@ -102,8 +101,20 @@ func New(log logr.Logger,
 // Start is a blocking func that will run the client facing certificate service
 func (s *Server) Start(ctx context.Context) error {
 	// Setup the grpc server using the passed TLS config
+	srvmetrics := grpcprom.NewServerMetrics(func(op *prom.CounterOpts) { op.Namespace = "cert_manager_istio_csr" })
+	srvmetrics.EnableHandlingTimeHistogram(func(op *prom.HistogramOpts) { op.Namespace = "cert_manager_istio_csr" })
 	creds := credentials.NewTLS(s.tls.Config())
-	grpcServer := grpc.NewServer(grpc.Creds(creds))
+	grpcServer := grpc.NewServer(
+		grpc.StreamInterceptor(srvmetrics.StreamServerInterceptor()),
```

Comment:
Question: do we need this interceptor? I think we don't stream anything

Reply:
https://github.com/cert-manager/istio-csr/pull/72/commits/b8641b0d291ae266dcd2f7f8f57e1cdae5b48044